src/dfx/src/lib/extension/install.rs

```diff
@@ -0,0 +1,34 @@
+use std::io::Cursor;
+use std::os::unix::fs::PermissionsExt;
+
+use crate::lib::extension::manifest::ExtensionLocation;
+use crate::lib::extension::manifest::ExtensionsManifest;
+use anyhow::Error;
+use semver::Version;
+
+use std::path::Path;
+
+pub fn install_extension(
+    extensions_dir: &Path,
+    dfx_version: Version,
+    extension_name: &str,
+) -> Result<(), Error> {
+    let manifest = ExtensionsManifest::fetch().unwrap();
+
+    if let Some(ExtensionLocation { download_url }) = manifest.find(extension_name, dfx_version) {
+        println!("installing: {} from {}", extension_name, &download_url);
+        let extension_dir = extensions_dir.join(extension_name);
+        let extension_archive = extension_dir.join(extension_name);
+        let response = reqwest::blocking::get(&download_url).unwrap();
+        std::fs::create_dir_all(extension_dir).unwrap();
```

Comment:
As written, if this function were interrupted after `create_dir_all` but before completing the rest of the work, we would have a partially installed extension, possibly with a corrupt / incomplete executable.

This should download/unpack the archive into a temporary directory.  This temporary directory should be on the same filesystem as the disk cache, and one way to do this is to have the temporary directory be a sibling of the final directory.

This way, after unpacking/copying files, we can install the extension atomically by renaming it.